### DISCUSSION TITLE
Issue while converting a clipped image and mask to a group

### DISCUSSION THREAD

**Discussion by nerd-at-hyd** [fabricjs/fabric.js#7997] [Q&A] [answered]
## Version
5.2.1

## Test Case
https://jsfiddle.net/k62sj9q3/1/

## Information about environment
Google Chrome

## Steps To Reproduce
Please have a look at the jsfiddle above, To recreate the issue, click on clip, then click on group button.

## Expected Behavior
It should not move the clipped image after grouping the clipped image with mask.  

## Actual Behavior
It's moving the clipped image away after grouping the clipped image with mask.

**Before clipping the image:**
![image](https://user-images.githubusercontent.com/73891255/172631665-b21f024e-f98b-4adc-a725-debe0a0a3495.png)

**Before Grouping the clipped image:**
![image](https://user-images.githubusercontent.com/73891255/172632054-986a7028-3e56-4f9c-869d-0d16d7c6e235.png)

**After Grouping the clipped image:**
![image](https://user-images.githubusercontent.com/73891255/172632340-08fbc1a4-b778-4bdd-b9f7-c8152ce19e71.png)

**Comment by ShaMan123** [comment, marked as answer]
It's an awful mistake to have an object act as an object and as a clip path of another object particularly when both are in the same group.

**Reply by ShaMan123** [reply]
look at princess mononoke
http://fabricjs.com/clipping

**Reply by ShaMan123** [reply]
and use `clone`

**Reply by nerd-at-hyd** [reply]
You were right, I am not supposed to use same object as an canvas object and as a clipPath, so if I skip adding the object to canvas and just use that as a clip path, it works, 
**BUT for some reason I need that object as a canvas object and same object used as a clipPath, so I cloned it as you said, but somehow clipping with clone doesn't work as expected, seems object cloning is not working properly, please check the [fiddle]**(https://jsfiddle.net/96o7cht3/)

```
function clip() {
	mask.clone(function(clone) {
  	   //clone.setCoords();
           pub.clipPath = clone;
           this.canvas.renderAll();
        });
}
```

**Reply by ShaMan123** [reply]
> look at princess mononoke
http://fabricjs.com/clipping

**Reply by nerd-at-hyd** [reply]
Hi @ShaMan123 
I am able achieve this, please check fiddle: https://jsfiddle.net/k62sj9q3/6/
**But when I resize the group it distort the image, I think it's happening due to `absolutePositioned`, but if I make `absolutePositioned `as false, the clipping itself doesn't work.**
The princess mononoke doesn't use `absolutePositioned` so resizing works as expected, but for me without `absolutePositioned`, clipping doesn't work. Please help.
